### PR TITLE
Faster division routines

### LINKED ISSUE #221: Rewrite FM/MOD
FM/MOD implementation is taken from Gforth.
Since Gforth is GPL V3, it is legally questionable to use its code in this MIT licensed project.

### PR DESCRIPTION
**Pull request by burnsauce**
Fixes #221 and #194

### CODE REVIEW

File: math.asm [hunk 1]
```diff
@@ -187,8 +188,8 @@ ABS
     rts
 
 DABS_STAR           ; ( n1 n2 -- ud1 )
-    lda MSB,x	    ;	ud1 = abs(n1) * abs(n2)
-    eor MSB+1,x     ;	with final sign output in A register
+    lda MSB,x      ;   ud1 = abs(n1) * abs(n2)
+    eor MSB+1,x     ;  with final sign output in A register
```
Comment (burnsauce) [inline math.asm:192]: Couldn't get it right in the merge somehow :(

**Comment by burnsauce** [comment]
This branch completes Integer Calculations benchmark in 27.7 seconds. :)

File: math.asm (hunk 1 above)
Comment (jkotlinski) [reply math.asm:192]: it's ok!

**Comment by jkotlinski** [comment]
wow! excellent work!

**Pull request merged**